apps/bynder/src/index.js

```diff
@@ -43,51 +66,90 @@ function makeThumbnail(resource) {
 }
 
 function prepareBynderHTML({ bynderURL, assetTypes }) {
-  let types = '';
-  if (!assetTypes) {
-    // We deault to just images in this fallback since this is the behavior the App had in its initial release
-    types = 'image';
-  } else {
-    types = assetTypes.trim().split(',').map(type => type.trim()).join(',');
-  }
-
   return `
     <div class="dialog-container">
-      <div
-        id="bynder-compactview"
-        data-assetTypes="${types}"
-        data-autoload="true"
-        data-button="Load media from bynder.com"
-        data-collections="true"
-        data-folder="bynder-compactview"
-        data-fullScreen="true"
-        data-header="false"
-        data-language="en_US"
-        data-mode="multi"
-        data-zindex="300"
-        data-defaultEnvironment="${bynderURL}"
-      />
-    </div>
+      <div id="bynder-compactview" />
+    </div>      
   `;
 }
 
+function transformAsset(asset) {
+  return ({
+    "id": asset.databaseId,
+    "orientation": asset.orientation.toLowerCase(),
+    "archive": asset.isArchived ? 1 : 0,
+    "type": asset.type.toLowerCase(),
+    "fileSize": asset.fileSize,
+    "description": asset.description,
+    "name": asset.name,
+    "height": asset.height,
+    "width": asset.width,
+    "copyright": asset.copyright,
+    "extension": asset.extensions,
+    "userCreated": asset.createdBy,
+    "datePublished": asset.publishedAt,
+    "dateCreated": asset.createdAt,
+    "dateModified": asset.updatedAt,
+    "watermarked": asset.isWatermarked ? 1 : 0,
+    "limited": asset.isLimitedUse ? 1 : 0,
+    "isPublic": asset.isPublic ? 1 : 0,
+    "brandId": asset.brandId,
+    "thumbnails": {
+      "contentful": asset.files.contentful.url,
```

Comment:
these properties on files aren't necessarily defined, which can make this throw an error when trying to add a file. Can we use optional chaining?

Reply:
I'm also worried that there might be other properties that used to exist on Thumbnails that we've missed here. Were you able to find a complete list anywhere?

Reply:
not really :/ That were the only ones I found. I changed it to be generated dynamically now, however for `webImage` and `thumbnail` I still had to manually map it, as the naming has changed and to support backwards compatibility. Do you think it's save now?